cmd/install.go

```diff
@@ -278,378 +272,427 @@ func runInstallDupeDetectionSubCommand(ctx context.Context, config *configs.Conf
+	defer log.WithContext(ctx).Infof("Install path is %s", installPath)
+
+	if err := utils.CreateFolder(ctx, installPath, config.Force); os.IsExist(err) {
+		reader := bufio.NewReader(os.Stdin)
+		log.WithContext(ctx).Warnf("%s. Do you want continue to install? Y/N", err.Error())
+		line, readErr := reader.ReadString('\n')
+		if readErr != nil {
+			log.WithContext(ctx).WithError(readErr).Error("Exiting...")
+			return readErr
+		}
+
+		if strings.TrimSpace(line) == "Y" || strings.TrimSpace(line) == "y" {
+			config.Force = true
+			if err = utils.CreateFolder(ctx, installPath, config.Force); err != nil {
+				log.WithContext(ctx).WithError(err).Error("Exiting...")
+				return err
+			}
+		} else {
+			log.WithContext(ctx).Warn("Exiting...")
+			return err
+		}
+	}
+
+	return nil
+}
+
+func checkInstalledPackages(ctx context.Context) (err error) {
+	// TODO: 1) must offer to install missing packages
+	// TODO: 2) add support for Windows and Mac
+	if utils.GetOS() == constants.Linux {
+		installedCmd := utils.GetInstalledPackages(ctx)
+		var notInstall []string
+		for _, p := range constants.DependenciesPackages {
+			if _, ok := installedCmd[p]; !ok {
+				notInstall = append(notInstall, p)
+			}
+		}
+
+		if len(notInstall) > 0 {
+			return errors.New(strings.Join(notInstall, ", ") + " is missing from your OS, which is required for running, please install them")
+		}
+	}
+	return nil
+}
+
 func installComponent(ctx context.Context, config *configs.Config, installCommand constants.ToolType, version string) (err error) {
 	commandName := strings.Split(string(installCommand), "/")[len(strings.Split(string(installCommand), "/"))-1]
 	log.WithContext(ctx).Infof("Installing %s executable...", commandName)
 
 	downloadURL, execArchiveName, err := config.Configurer.GetDownloadURL(version, installCommand)
 	if err != nil {
-		return errors.Errorf("failed to get download url, err: %s", err)
+		log.WithContext(ctx).WithError(err).Error("failed to get download url")
+		return err
 	}
 
 	if err = installExecutable(ctx, config, downloadURL.String(), execArchiveName, installCommand); err != nil {
-		log.WithContext(ctx).Errorf("Install %s executable failed", commandName)
+		log.WithContext(ctx).WithError(err).Errorf("Install %s executable failed", commandName)
 		return err
 	}
 
 	log.WithContext(ctx).Infof("%s executable installed successfully", commandName)
 
-	if installCommand == constants.PastelD {
-		if err = InitCommandLogic(ctx, config); err != nil {
-			log.WithContext(ctx).Error("Initialize the node")
+	return nil
+}
+
+func installExecutable(ctx context.Context, config *configs.Config,
+	downloadURL string,
+	archiveName string,
+	toolType constants.ToolType) (err error) {
+
+	if err = utils.DownloadFile(ctx, filepath.Join(config.PastelExecDir, archiveName), downloadURL); err != nil {
+		log.WithContext(ctx).Errorf(fmt.Sprintf("Failed to download pastel executable file : %s", downloadURL))
```

Comment:
could you change:
`log.WithContext(ctx).Errorf(fmt.Sprintf("Failed to download pastel executable file : %s", downloadURL))`
to 
`log.WithContext(ctx).WithError(err).Errorf("Failed to download pastel executable file : %s", downloadURL)`

Reply:
yes, missed this on